modules/storage-api/src/main/java/org/apache/ignite/internal/storage/MvPartitionStorage.java

```diff
@@ -225,8 +225,9 @@ public interface MvPartitionStorage extends ManuallyCloseable {
      * @param lowWatermark A time threshold for the row. Rows younger then the watermark value will not be removed.
      * @return A pair of table row and row id, where a timestamp of the row is less than or equal to {@code lowWatermark}.
      *      {@code null} if there's no such value.
+     * @throws StorageException If failed to poll element for vacuum.
      */
-    default @Nullable TableRowAndRowId pollForVacuum(HybridTimestamp lowWatermark) {
+    default @Nullable TableRowAndRowId pollForVacuum(HybridTimestamp lowWatermark) throws StorageException {
```

Comment:
`StorageException` is not checked, there is no need to write it in the signature, it is enough to mention it in the documentation.

Reply:
Every other method of MvPartitionStorage explicitly states that it throws StorageException, so this looks consistent

Reply:
These methods do not have an `StorageException` in their signature:
- 'org.apache.ignite.internal.storage.MvPartitionStorage#flush'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#lastAppliedIndex'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#lastAppliedTerm'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#persistedIndex'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#committedGroupConfiguration()'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#committedGroupConfiguration(org.apache.ignite.internal.storage.RaftGroupConfiguration)'
- 'org.apache.ignite.internal.storage.MvPartitionStorage#close'

Reply:
What about:
- `V execute() throws StorageException;`
- `<V> V runConsistently(WriteClosure<V> closure) throws StorageException`
- `void lastApplied(long lastAppliedIndex, long lastAppliedTerm) throws StorageException`
- `ReadResult read(RowId rowId, HybridTimestamp timestamp) throws StorageException;`
- `@Nullable TableRow addWrite(RowId rowId, @Nullable TableRow row, UUID txId, UUID commitTableId, int commitPartitionId) throws TxIdMismatchException, StorageException` (this one has two!)
- `@Nullable TableRow abortWrite(RowId rowId) throws StorageException`
- `void commitWrite(RowId rowId, HybridTimestamp timestamp) throws StorageException`
- `void addWriteCommitted(RowId rowId, @Nullable TableRow row, HybridTimestamp commitTimestamp) throws StorageException`
- `Cursor<ReadResult> scanVersions(RowId rowId) throws StorageException`
- `PartitionTimestampCursor scan(HybridTimestamp timestamp) throws StorageException;`
- `@Nullable RowId closestRowId(RowId lowerBound) throws StorageException;`

Reply:
You wrote "Every other", but it's not, I suggest not to change the signature.